app/scripts/components/exploration/analysis-data.ts

```diff
@@ -246,20 +246,37 @@ export async function requestDatasetTimeseriesData({
         timeseries: layerStatistics
       }
     });
+    return {
+      status: TimelineDatasetStatus.SUCCESS,
+      meta: {
+        total: assets.length,
+        loaded: assets.length
+      },
+      error: null,
+      data: {
+        timeseries: layerStatistics
+      }
+    };
   } catch (error) {
     // Discard abort related errors.
-    if (error.revert) return;
+    if (error.revert) {
+      return {
+        status: TimelineDatasetStatus.LOADING,
+        error: null,
+        data: null,
+        meta: {}
+      };
+    }
 
     // Cancel any inflight queries.
     queryClient.cancelQueries({ queryKey: ['analysis', id] });
     // Remove other requests from the queue.
     concurrencyManager.dequeue(`${id}-analysis-asset`);
-
-    onProgress({
+    return {
       ...datasetAnalysis,
       status: TimelineDatasetStatus.ERROR,
       error,
       data: null
-    });
+    };
```

Comment:
💭 : hmm.. I see that in some places we are returning type `TimelineDatasetAnalysis` for it to be later set to the analysis and then in others, we are invoking the callback. i'm wondering a few things to clean this up.. 

1. can we just pass in `setAnalysis` directly, no need for `onProgress`
2. define a var of type `TimelineDatasetAnalysis` and have it overwrite and then just return that var or pass it to `setAnalysis`. this is really a nit in term of it looking a little more cleaner

What do you think?